### PR TITLE
fix: switch to momentjs to handle timezones

### LINKED ISSUE #428: Use all Kibana timezones in the timeZone prop
**Is your feature request related to a problem? Please describe.**
the `timeZone` prop use only IANA time zones identifiers.
Kibana use instead a mix of IANA and other time zone indentifiers.

**Describe the solution you'd like**
We should support all time zones available in kibana

**Describe alternatives you've considered**
n/a

**Additional context**
n/a

**Kibana Cross Issues**
n/a

**Checklist**

- [x] this request is checked against already exist requests
- ~[ ] every related Kibana issue is listed under `Kibana Cross Issues` list~
- ~[ ] `kibana cross issue` tag is associated to the issue if any kibana cross issue is present~

### PR DESCRIPTION
**Pull request by markov00**
## Summary

The current used time library, `Luxon` is using `Intl.DateTimeFormatter` as the main source for handling time zones. That library depends on browsers implementation. IE11 doesn't have any timezone and it will need a polyfill.
This PR replace `luxon` with `momentjs` and `moment-timezone`. I'm leaving `luxon` as a devDependency on the tests and on stories. We can remove it in a future phase.

`momentjs` and `moment-timezone` are configured as `peerDependency` since Kibana is already using that library.

I've checked that fix cross-linking the build with Kibana. I've tested various timezones in Discover chart.

fix #428

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

- [ ] Any consumer-facing exports were added to `src/index.ts` (and stories only import from `../src` except for test data & storybook)
- [x] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)
- [ ] Proper documentation or storybook story was added for features that require explanation or tutorials
- [ ] Unit tests were updated or added to match the most common scenarios
- [x] Each commit follows the [convention](https://github.com/elastic/elastic-charts/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Comment by markov00** [comment]
to be merged after https://github.com/elastic/elastic-charts/pull/430

File: package.json [hunk 1]
```diff
@@ -127,15 +132,13 @@
   },
   "dependencies": {
     "@types/d3-shape": "^1.3.1",
-    "@types/luxon": "^1.11.1",
```
[the remark this answers is not in the document]
Comment (markov00) [reply package.json]: @nickofthyme if we have `moment-timezone` as peerDependency, should we add the types as peer?

[the remark this answers is not in the document]
Comment (nickofthyme) [reply package.json]: idk, kibana has `@types/moment-timezone` as a devDep so I think it should be fine. I think the difference between luxon and moment is that moment has the types built into the bundle, not as a separate definitely typed package.

File: yarn.lock [hunk 2]
```diff
@@ -10704,6 +10711,18 @@ modify-values@^1.0.0:
   resolved "https://registry.yarnpkg.com/modify-values/-/modify-values-1.0.1.tgz#b3939fa605546474e3e3e3c63d64bd43b4ee6022"
   integrity sha512-xV2bxeN6F7oYjZWTe/YPAy6MN2M+sL4u/Rlm2AHCIVGfo2p1yGmBHQ6vHehl4bRTZBdHu3TSkWdYgkwpYzAGSw==
 
+moment-timezone@^0.5.27:
+  version "0.5.27"
+  resolved "https://registry.yarnpkg.com/moment-timezone/-/moment-timezone-0.5.27.tgz#73adec8139b6fe30452e78f210f27b1f346b8877"
+  integrity sha512-EIKQs7h5sAsjhPCqN6ggx6cEbs94GK050254TIJySD1bzoM5JTYDwAU1IoVOeTOL6Gm27kYJ51/uuvq1kIlrbw==
+  dependencies:
+    moment ">= 2.9.0"
```
[the remark this answers is not in the document]
Comment (markov00) [reply yarn.lock]: solved in c52c934a26e0e637729b57dc6bc562e7b9d22498

File: src/utils/data/formatters.ts [hunk 3]
```diff
@@ -29,7 +24,7 @@ export function niceTimeFormatByDay(days: number) {
     return 'MMMM dd';
   }
   if (days > 1 && days <= 7) {
-    return 'MM-dd HH:mm';
+    return 'MM-dd HH:mm ';
```
[the remark this answers is not in the document]
Comment (markov00) [reply src/utils/data/formatters.ts]: fixed in https://github.com/elastic/elastic-charts/pull/436/commits/0460c4f1b3788130aa1d28b8ea4dc66a66edfeb6

File: src/utils/data/formatters.ts [hunk 4]
```diff
@@ -1,26 +1,21 @@
-import { DateTime, Interval } from 'luxon';
+import { TickFormatter, TickFormatterOptions } from '../../chart_types/xy_chart/utils/specs';
+import { getMomentWithTz } from './date_time';
+import moment from 'moment';
```
[the remark this answers is not in the document]
Comment (markov00) [reply src/utils/data/formatters.ts]: fixed in https://github.com/elastic/elastic-charts/pull/436/commits/0460c4f1b3788130aa1d28b8ea4dc66a66edfeb6

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (markov00) [reply package.json]: yep, that should work. Let's build a real package and see if Kibana complains
I've tested it locally and seems fine

File: package.json [hunk 5]
```diff
@@ -106,6 +108,9 @@
     "jest-puppeteer": "^4.3.0",
     "jest-puppeteer-docker": "^1.2.0",
     "lorem-ipsum": "^2.0.3",
+    "luxon": "^1.11.3",
+    "moment": "^2.24.0",
```
[the remark this answers is not in the document]
Comment (markov00) [reply package.json]: done in https://github.com/elastic/elastic-charts/pull/436/commits/c52c934a26e0e637729b57dc6bc562e7b9d22498

**Comment by markov00** [comment]
jenkins test this please

**Comment by nickofthyme** [comment]
retest

**Comment by nickofthyme** [comment]
jenkins test this please

**Comment by markov00** [comment]
jenkins test this please

**Pull request merged**